server/src/test/java/org/eclipse/lsp/cobol/extendedapi/CFASTBuilderTest.java

```diff
@@ -0,0 +1,133 @@
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.reader.CobolLineReader;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.reader.CobolLineReaderImpl;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.rewriter.CobolCommentEntriesMarkerImpl;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.rewriter.CobolInlineCommentEntriesNormalizerImpl;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.rewriter.CobolLineIndicatorProcessorImpl;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.rewriter.CobolLineReWriter;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.transformer.CobolLinesTransformation;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.transformer.ContinuationLineTransformation;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.ReplacingService;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.ReplacingServiceImpl;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.writer.CobolLineWriter;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.writer.CobolLineWriterImpl;
+import org.eclipse.lsp.cobol.core.semantics.SemanticContext;
+import org.eclipse.lsp.cobol.core.strategy.CobolErrorStrategy;
+import org.eclipse.lsp.cobol.jrpc.CobolLanguageClient;
+import org.eclipse.lsp.cobol.service.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Test for @link({@link org.eclipse.lsp.cobol.service.CFASTBuilderImpl}.
+ */
+@Slf4j
+public class CFASTBuilderTest {
+  private static Gson gson = new GsonBuilder().setPrettyPrinting().create();
+  static Stream<Arguments> casesToTest() throws IOException {
+    return Files.list(Paths.get("src", "test", "resources", "cfast"))
+            .filter(p -> p.toString().endsWith(".cbl"))
+            .map(p -> {
+              try {
+                final String resultName = p.getFileName().toString().replace(".cbl", ".result.json");
+                final String resultContent = new String(Files.readAllBytes(p.getParent().resolve(resultName)));
+                return Arguments.arguments(new String(Files.readAllBytes(p)), resultContent, p.getFileName().toString());
+              } catch (IOException e) {
+                LOG.error(e.getMessage(), e);
+              }
+              return null;
+            });
+  }
+
+  private CobolLanguageEngine engine;
+
+  /**
+   * Initialize cobol engine.
+   */
+  @BeforeEach
+  public void initEngine() {
+    MessageService messageService = mock(MessageService.class);
```

Comment:
Why don't you start up the Guice context?

Reply:
I have no idea. Do you think it will make more sense here?

Reply:
Definitely. It also means that you should not expose all the used constructors.